### PR TITLE
build: add update-go-format.sh to do go format

### PR DESCRIPTION
**Pull request by Fish-pro**
Signed-off-by: Fish-pro <zechun.chen@daocloud.io>

#### What type of PR is this?

/kind feature

#### Which issue(s) this PR fixes:

Part of #232

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: hack/update-go-format.sh [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+REPO_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
+source "${REPO_ROOT}"/hack/util.sh
+
+util::verify_go_version
```
Comment (wzshiming) [inline hack/update-go-format.sh:23]:
```suggestion
```

It seems that `go` is not used here, so there should be no need to check if it exists

File: hack/update-go-format.sh [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+cd "${REPO_ROOT}"
+find_files() {
```
Comment (wzshiming) [inline hack/update-go-format.sh:26]:
```suggestion
function find_files() {
```

File: hack/update-go-format.sh [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  find . -not \( \
+      \( \
+        -wholename './bin' \
+        -o -wholename './.git' \
+        -o -wholename './site/*' \
+        -o -wholename '*/vendor/*' \
+      \) -prune \
+    \) -name '*.go'
+}
```
Comment (wzshiming) [inline hack/update-go-format.sh:35]: The go files only exist under `cmd` and `pkg`, so maybe we should just find these two directories, and change them together in `verify-go-format.sh`

File: hack/update-go-format.sh (hunk 1 above)
Comment (Fish-pro) [reply hack/update-go-format.sh:23]: I understand that `gofmt` depends on the `go` installation, so it needs to be verified
image

Comment (wzshiming) [reply hack/update-go-format.sh:23]: `gofmt` is a standalone binary file and does not depend on `go`

https://github.com/golang/go/blob/master/src/cmd/gofmt/gofmt.go

Comment (Fish-pro) [reply hack/update-go-format.sh]: Thanks for tips

File: hack/verify-go-format.sh [hunk 4]
```diff
@@ -21,7 +21,7 @@ ROOT_DIR="$(dirname "${BASH_SOURCE[0]}")/.."
 
 function check() {
   echo "Verify gofmt"
-  out="$(gofmt -l -d $(find . -name '*.go'))"
+  out="$(gofmt -l -d $(find pkg -name '*.go'))"
```
Comment (wzshiming) [inline hack/verify-go-format.sh:24]: What about `cmd`?

Comment (Fish-pro) [reply hack/verify-go-format.sh:29]: Sorry, It's not saved. It's local

File: hack/verify-go-format.sh [hunk 5]
```diff
@@ -21,7 +21,12 @@ ROOT_DIR="$(dirname "${BASH_SOURCE[0]}")/.."
 
 function check() {
   echo "Verify gofmt"
-  out="$(gofmt -l -d $(find . -name '*.go'))"
+  out="$(gofmt -l -d $(find cmd -name '*.go'))"
+  if [[ -n "${out}" ]]; then
+    echo "${out}"
+    return 1
+  fi
+  out="$(gofmt -l -d $(find pkg -name '*.go'))"
```
Comment (wzshiming) [inline hack/verify-go-format.sh:29]:
```suggestion
  out="$(gofmt -l -d $(find cmd pkg -name '*.go'))"
```

File: hack/update-go-format.sh [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+
+REPO_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
+
+cd "${REPO_ROOT}"
+function find_files() {
+  find cmd -name '*.go'
+  find pkg -name '*.go'
+}
+
+find_files | xargs gofmt -s -w
```
Comment (wzshiming) [inline hack/update-go-format.sh:28]:
```suggestion
gofmt -s -w $(find cmd pkg -name '*.go')
```

**Review by wzshiming — APPROVED** [review]
/approve
/lgtm

**Pull request merged**